Add basic block inserter features: grid, search, icons, ordering

kean opened the pull request:
## Changes 

- Add `BlockInserterView` and a set or related types to render the blocks in a grid (closes [CMM-857](https://linear.app/a8c/issue/CMM-857/show-blocks-in-a-grid))
- Add context menus for blocks (closes [CMM-863](https://linear.app/a8c/issue/CMM-863/add-context-menu-and-details-view-for-blocks))
- Make the inserter searchable (closes [CMM-858](https://linear.app/a8c/issue/CMM-858/make-block-inserter-searchable))

## Testing Instructions

- Enable "Native Inserter"
- Tap "+"
- **Verify** that the new inserter is shown
- **Verify** that you can long-press on a block to reveal more details
- **Verify** that you can search for a block using it's name, title, keywords or category (search weights defined in that order) 
- **Verify** that Dynamic Type and Dark Mode are supported

**Note**: you can not yet tap to select a block.

## Technical Notes

I added [SVGKit](https://github.com/SVGKit/SVGKit) to render block icons. It seems like the only viable option.

## Limitations

- `BlockInserterViewModel` establishes _some_ mobile-specific ordering for blocks, but I opened a separate ticket ([CMM-875](https://linear.app/a8c/issue/CMM-875/establish-the-mobile-specific-order-for-blocks)) to further enhance it (see screenshots/comments). I'm considering implementing it on the JS level.
- There is some non-localized text in this PR. I opened [CMM-874](https://linear.app/a8c/issue/CMM-874/add-localization-for-category-names-etc).
- Some block icons contain incorrect SVG ([CMM-873](https://linear.app/a8c/issue/CMM-873/svg-for-icons-for-some-blocks-is-invalid)). I'm considering fixing it at the source and/or patching them in GBK.

## Screenshots

<img width="360" alt="Screenshot 2025-10-21 at 12 59 49 PM" src="https://github.com/user-attachments/assets/0a7d1fad-b9fb-457f-a1ea-4138ed5b22cb" /> <img width="360" alt="Screenshot 2025-10-21 at 1 02 41 PM" src="https://github.com/user-attachments/assets/db1c9d02-e606-486e-9932-0f91e88449e7" /> 
<img width="360" alt="Screenshot 2025-10-21 at 1 02 46 PM" src="https://github.com/user-attachments/assets/8fa96dba-767a-4e6b-8ccb-3c7cbd1e3025" />  <img width="360" alt="Screenshot 2025-10-21 at 1 04 00 PM" src="https://github.com/user-attachments/assets/b8c5aa10-3775-411c-87d8-8a8b5cfa34da" />

kean committed:
Add basic block inserter features: show blocks, search

kean committed:
Add dark mode support

kean committed:
Update logging

kean committed:
Remove duplicated toolbar content

kean committed:
Remove sortTextBlocks

kean committed:
Simplify ordering for categories

kean committed:
Fix crash when showing BlockInserterBlockView

kean committed:
Make EditorBlock just Decodable again

kean committed:
Cleanup searchableFields

kean commented on ios/Sources/GutenbergKit/Sources/Helpers/SearchEngine.swift:
```diff
@@ -0,0 +1,190 @@
+import Foundation
+
+/// A protocol for items that can be searched
+protocol Searchable {
```
**Disclaimer**: other code in PR is written by hand, but this file is entirely generated. It uses Levenshtein distance for fuzzy matching, and does scoring depending on weight of searchable fields. It seems to work pretty well, and we do need it in the inserter.

dcalhoun replied:
Searching worked relatively well from my testing. I noted one thing while searching:  we are not honoring the [`inserter`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#inserter) supports attribute for blocks. Blocks setting this to `false` should not appear in the block inserter or its search results.

kean committed:
Fewer blocks in the preview

kean committed:
Move getSerializedBlocks to blocks.js

kean commented on src/utils/blocks.js:
```diff
@@ -24,3 +25,65 @@ export function unregisterDisallowedBlocks( allowedBlockTypes ) {
+ *
+ * @return {string|null} The serialized icon string or null.
+ */
+function getBlockIcon( blockType ) {
```
OK, I generated this too. I might need some help verifying if it makes sense (it works).

dcalhoun replied:
Yes, it makes sense. It seems necessary given the desired outcome of providing SVG strings to the native host app. I cannot think of a better alternative or specific problems with the approach at this time.

kean committed:
Fix lint errors

dcalhoun commented on src/utils/bridge.js:
```diff
@@ -93,15 +93,7 @@ export function onBlocksChanged( isEmpty = false ) {
-			keywords: blockType.keywords || [],
-		};
-	} );
+	const blocks = getSerializedBlocks();
```
I haven't reviewed all the code yet, but wanted to note we need to rebase to include #198 and adapt this implementation so that the remote editor works. 

Good news is that the native inserter does work with the remote editor in my quick merge conflict resolution on my local machine.

kean committed:
Add missing async

dcalhoun committed:
fix: Defer remote editor references to `@wordpress` module globals

The static import resulted in `window.wp` references prior them being
defined. This problem is unique to the remote editor, which
asynchronously loads site-specific `@wordpress` module scripts. Using a
dynamic import within the function ensures the reference does occur
until after the globals are defined.

dcalhoun commented on src/utils/blocks.js:
```diff
@@ -24,3 +25,63 @@ export function unregisterDisallowedBlocks( allowedBlockTypes ) {
+ */
+export function getSerializedBlocks() {
+	return getBlockTypes().map( ( blockType ) => {
+		return {
+			name: blockType.name,
+			title: blockType.title,
+			description: blockType.description,
+			category: blockType.category,
+			keywords: blockType.keywords || [],
+			icon: getBlockIcon( blockType ),
+		};
```
As noted in my comment regarding search, we likely need to include the [`inserter`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#inserter) supports attribute so that we can honor that setting.

kean replied:
I added it as a [bug](https://linear.app/a8c/issue/CMM-888/add-support-for-the-inserter-attribute) and will address in the upcoming PRs. Thanks!

dcalhoun reviewed (approved):
Approving as this works well from my testing and other PRs build atop this. My inline comments note one issue we should address regarding inserter block inclusion.

dcalhoun commented:
Noting that this may be a breaking change for `WordPress-iOS`. When we integrate the latest GBK changes, we'll need to address this:

```
/Library/Developer/Xcode/DerivedData/WordPress-bwtwzztzxastvrfeguwfttopcahk/SourcePackages/checkouts/SVGKit/Source/include/Comment.h:17:12 'Comment' has different definitions in different modules; first difference is definition in module 'SVGKit' found super class with type 'CharacterData'
```
